### PR TITLE
6.0-migration-guide.adoc, 6.0-release-notes.adoc & 23 questions docum…

### PR DESCRIPTION
**Pull request by brian-forte**
…ents added

All files added to release-notes-DOC-1559 branch in `/modules/ROOT/pages/`.

Related Ticket: 

Description of Changes:
* Placeholder text

Pre-checks:
- [ ] Branch prefixed with `feature/` or `hotfix/`
- [ ] `_data/nav.yml` has been updated (if applicable)
- [ ] Files has been included where required (if applicable)
- [ ] Files removed have been deleted, not just excluded from the build (if applicable)
- [ ] (New product features only) Release Note added

Review:
- [ ] Documentation Team Lead has reviewed
- [ ] Product Manager has reviewed

### CODE REVIEW

File: modules/ROOT/pages/questions/question-01.adoc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+1. For each plugin re-classified as premium, the following sentence (and link) was used to denote this:
+
+   *{plugin-name} is now a https://tiny.cloud/tinymce/features/#productivity[Premium plugin].*
+
+   Is this desirable, or does it read too much like marketing?
+   
+   Is there, perhaps, a better URL to link out with?
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-01.adoc:7]: @EkimChau

File: modules/ROOT/pages/questions/question-02.adoc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+2. I’ve tentatively noted each ‘Plugin moved to Premium’ in the
+   *Migration Guide* with an admonishment that
+
+   *If you are 1) using the Free version of {productname} and 2) rely
+   on the `{plugin-name}` plugin, upgrading to {productname} 6.0 is
+   not recommended.*
+
+   Is this correct?
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-02.adoc:8]: @EkimChau

File: modules/ROOT/pages/questions/question-03.adoc [hunk 3]
```diff
@@ -0,0 +1,16 @@
+3. https://ephocks.atlassian.net/browse/TINY-8333[TINY-8333] (removing
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-03.adoc:1]: @TheSpyder

File: modules/ROOT/pages/questions/question-04.adoc [hunk 4]
```diff
@@ -0,0 +1,13 @@
+4. The `toc` plugin has been documented as _new and improved_ because
+   of the following:
+
+   *the `toc` plugin has modernised its output. It now uses the ES2020
+   library and targets ES2018. As well, it now uses
+   https://terser.org[terser] to improve performance.*
+
+   It felt like at least some explanation for moving the plugin to
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-04.adoc:8]: @lostkeys

File: modules/ROOT/pages/questions/question-04.adoc [hunk 5]
```diff
@@ -0,0 +1,13 @@
+4. The `toc` plugin has been documented as _new and improved_ because
+   of the following:
+
+   *the `toc` plugin has modernised its output. It now uses the ES2020
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-04.adoc:4]: @lnewson

File: modules/ROOT/pages/questions/question-02.adoc (hunk 2 above)
Comment (TheSpyder) [reply modules/ROOT/pages/questions/question-02.adoc:8]: It's definitely not correct. We will always recommend upgrading to the latest release, these plugins are still available just not for free.

File: modules/ROOT/pages/questions/question-03.adoc (hunk 3 above)
Comment (TheSpyder) [reply modules/ROOT/pages/questions/question-03.adoc:1]: The plugin is functionally identical, so we shouldn't need to say anything other than it moved to a premium feature.

**Comment by tiny-james** [comment]
Is there a link to this PR on the staging site? It's hard to review the tables of data in the markdown-like format.

File: modules/ROOT/pages/questions/question-05.adoc [hunk 6]
```diff
@@ -0,0 +1,12 @@
+5. In *TinyMCE 5.x*, the `paste_data_images` option was set to `false`
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-05.adoc:1]: @lnewson TINY-8310

File: modules/ROOT/pages/questions/question-04.adoc (hunk 5 above)
Comment (lnewson) [reply modules/ROOT/pages/questions/question-04.adoc:4]: This is a generic change done across all plugins and core. It's just that we no longer support IE 11 so we've modernized the output/requirements to run TinyMCE. It now needs a browser that's capable of running ES2020 which is actually all our supported browsers.

So basically this isn't something we need to mention in the release notes for Table of Contents specifically, it's a much larger change that impacts all of TinyMCE 6.0.

File: modules/ROOT/pages/questions/question-06.adoc [hunk 7]
```diff
@@ -0,0 +1,12 @@
+6. `paste` no longer supports Microsoft Word.
+
+   What precisely does this mean? If I copy styled text from Microsoft
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-06.adoc:3]: @EkimChau

File: modules/ROOT/pages/questions/question-07.adoc [hunk 8]
```diff
@@ -0,0 +1,14 @@
+7. two re-namings happened consequent to `noneditable` being re-cast
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-07.adoc:1]: @lnewson

File: modules/ROOT/pages/questions/question-08.adoc [hunk 9]
```diff
@@ -0,0 +1,12 @@
+8. A re-naming happened consequent to `textpatterns` being re-cast
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-08.adoc:1]: @metricjs TINY-8312

File: modules/ROOT/pages/questions/question-04.adoc (hunk 4 above)
Comment (TheSpyder) [reply modules/ROOT/pages/questions/question-04.adoc:8]: Table of contents and imagetools both have no user-facing changes in this release. We simply lifted the open source plugin and made it premium.

File: modules/ROOT/pages/questions/question-09.adoc [hunk 10]
```diff
@@ -0,0 +1,14 @@
+9. It’s noted in https://ephocks.atlassian.net/browse/TINY-8312[TINY-8312]
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-09.adoc:1]: @metricjs TINY-8312

File: modules/ROOT/pages/questions/question-10.adoc [hunk 11]
```diff
@@ -0,0 +1,18 @@
+10. the following plugins were removed from _TinyMCE 6.0_.
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-10.adoc:1]: @metricjs @EkimChau

File: modules/ROOT/pages/questions/question-11.adoc [hunk 12]
```diff
@@ -0,0 +1,13 @@
+11. the following plugins were removed from _TinyMCE 6.0_.
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-11.adoc:1]: @EkimChau @metricjs

File: modules/ROOT/pages/questions/question-05.adoc (hunk 6 above)
Comment (lnewson) [reply modules/ROOT/pages/questions/question-05.adoc:1]: > And, if so, should this change in the `paste_data_images` default
   be included in the *Migration Guide*?

Yes, this will need mentioning in the migration guide. 

> When upgrading to 6.0, `paste` goes from being a plugin to being
   Core functionality. Does this, as a consequence, cause the default
   to be changed?

This is not related to it moving to core. This was a separate improvement since it's often enabled anyways. The other reason if I recall correctly for this change was to bring it into line with PowerPaste which does allow local images to be pasted by default. @metricjs might be able to remember more there.

File: modules/ROOT/pages/questions/question-12.adoc [hunk 13]
```diff
@@ -0,0 +1,16 @@
+12. The `media` plugin’s `media_scripts` option, which was deprecated
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-12.adoc:1]: @lnewson @metricjs

File: modules/ROOT/pages/questions/question-06.adoc (hunk 7 above)
Comment (TheSpyder) [reply modules/ROOT/pages/questions/question-06.adoc:3]: We removed specific handling for Microsoft Word in this release. Content can still be pasted, and in fact it will appear with full formatting, but the HTML generated will be a mess. That's what our `paste` handling does; it makes the HTML nice.

There are two things worth mentioning in the release notes or migration guide:
- customers who wish to paste content from Microsoft Word should look at our premium paste offering
- we are offering the word import functionality to the community, if anyone can commit to maintaining it we will link to their plugin in our migration guide https://github.com/tinymce/tinymce/discussions/7487

File: modules/ROOT/pages/questions/question-07.adoc (hunk 8 above)
Comment (TheSpyder) [reply modules/ROOT/pages/questions/question-07.adoc:1]: This does need to be noted in the migration guide, but the settings are purely renamed. There are no functional changes.

Comment (lnewson) [reply modules/ROOT/pages/questions/question-07.adoc:1]: >  Socratic question: are these re-named options *Functions* or
   *Commands*?

Neither, these are just plain *Options* which take a string with the class name. For example, `noneditable_class: 'mceNonEditable'`.

> Do these re-namings constitute something an upgrader needs to be
   aware of? That is, does the re-naming need to be noted in the
   _Migration Guide_? And, if so, what does the upgrader need to do to
   cater to the change?

Yes, they'll also need to rename the options used in their TinyMCE init configuration to match the new names.

File: modules/ROOT/pages/questions/question-10.adoc (hunk 11 above)
Comment (TheSpyder) [reply modules/ROOT/pages/questions/question-10.adoc:1]: They were announced some time ago, left in place during the final few versions of 5 and are finally removed in 6
https://www.tiny.cloud/docs/release-notes/6.0-upcoming-changes/

File: modules/ROOT/pages/questions/question-11.adoc (hunk 12 above)
Comment (TheSpyder) [reply modules/ROOT/pages/questions/question-11.adoc:1]: No alternative:
* `fullpage`
* `bbcode`
* `legacyoutput`

Premium offering:
* `spellchecker`



These features are now built in, in fact I think all were built-in on 5 and left as empty plugins?
* `colorpicker`
* `contextmenu`
* `textcolor`

File: modules/ROOT/pages/questions/question-12.adoc (hunk 13 above)
Comment (lnewson) [reply modules/ROOT/pages/questions/question-12.adoc:1]: > a. Does anyone know what the setting did?

Based on the code it looks like it was meant to be a way to provide an allow list of `<script>`s that could be used to embed media. However, somewhere along the lines that seems to have broke and no longer works. That's only a guess though, as it was created before I started at Tiny and there's little documentation and no tests for it. @metricjs might know more...

> b. Is it safe to just document the option’s removal, leaving
    discovery of whatever (currently unknown) functionality is lost to
    the end-user(s) who was, previously, making use of said unknown
    capability?

This seems fine to me, but also not really my question to answer. @EkimChau is probably the best one to answer that.

File: modules/ROOT/pages/questions/question-11.adoc (hunk 12 above)
Comment (lnewson) [reply modules/ROOT/pages/questions/question-11.adoc:1]: > These features are now built in, in fact I think all were built-in on 5 and left as empty plugins?

This is correct, they are all part of core now.

File: modules/ROOT/pages/questions/question-13.adoc [hunk 14]
```diff
@@ -0,0 +1,14 @@
+13. According to https://ephocks.atlassian.net/browse/DOC-1533[DOC-1533]:
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-13.adoc:1]: @lnewson TINY-8382

File: modules/ROOT/pages/questions/question-14.adoc [hunk 15]
```diff
@@ -0,0 +1,9 @@
+14. Inferring from
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-14.adoc:1]: @lnewson 

FYI: https://www.tiny.cloud/docs/api/tinymce.html/tinymce.html.schema/

File: modules/ROOT/pages/questions/question-15.adoc [hunk 16]
```diff
@@ -0,0 +1,9 @@
+15. According to https://ephocks.atlassian.net/browse/DOC-1533[DOC-1533]:
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-15.adoc:1]: @lnewson TINY-8382

File: modules/ROOT/pages/questions/question-16.adoc [hunk 17]
```diff
@@ -0,0 +1,12 @@
+16. Options in the `link` plugin were re-named. To wit:
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-16.adoc:1]: @HAFRMO TINY-4603

File: modules/ROOT/pages/questions/question-17.adoc [hunk 18]
```diff
@@ -0,0 +1,13 @@
+17. https://ephocks.atlassian.net/browse/TINY-4603[TINY-4603] notes
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-17.adoc:1]: @HAFRMO @ltrouton TINY-4603

File: modules/ROOT/pages/questions/question-18.adoc [hunk 19]
```diff
@@ -0,0 +1,12 @@
+18. https://ephocks.atlassian.net/browse/TINY-4603[TINY-4603] notes
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-18.adoc:1]: @HAFRMO @ltrouton

File: modules/ROOT/pages/questions/question-19.adoc [hunk 20]
```diff
@@ -0,0 +1,31 @@
+19. A state of gentle confusion as much as a question.
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-19.adoc:1]: @ltrouton @metricjs

File: modules/ROOT/pages/questions/question-20.adoc [hunk 21]
```diff
@@ -0,0 +1,25 @@
+20. I’ve documented the https://ephocks.atlassian.net/browse/TINY-8104[TINY-8104]
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-20.adoc:1]: @HAFRMO @lnewson

File: modules/ROOT/pages/questions/question-21.adoc [hunk 22]
```diff
@@ -0,0 +1,23 @@
+21. `mceInsertTable`` -> `mceInsertTableDialog`
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-21.adoc:1]: @lnewson

File: modules/ROOT/pages/questions/question-22.adoc [hunk 23]
```diff
@@ -0,0 +1,38 @@
+22. Two elements default values were changed:
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-22.adoc:1]: @ltrouton TINY-8259

File: modules/ROOT/pages/questions/question-23.adoc [hunk 24]
```diff
@@ -0,0 +1,23 @@
+23. Another state of gentle confusion as much as a question.
```
Comment (EkimChau) [inline modules/ROOT/pages/questions/question-23.adoc:1]: @ltrouton

File: modules/ROOT/pages/questions/question-21.adoc (hunk 22 above)
Comment (lnewson) [reply modules/ROOT/pages/questions/question-21.adoc:1]: This wasn't related to my change, so tagging the right person: @ltrouton

File: modules/ROOT/pages/questions/question-13.adoc (hunk 14 above)
Comment (lnewson) [reply modules/ROOT/pages/questions/question-13.adoc:1]: This is not correct. The SaxParser linked is a Java implementation and not something we can use. The SaxParser that was used by the media plugin was the one in the public TinyMCE APIs: https://www.tiny.cloud/docs/api/tinymce.html/tinymce.html.saxparser/

It is now still using a TinyMCE public API, it's just our `DomParser` API instead since we've removed the custom HTML SaxParser implementation in 6.0 (https://www.tiny.cloud/docs/api/tinymce.html/tinymce.html.domparser/ is the equivalent from 5.x). So the change here is that the contents within media elements must be configured via the TinyMCE Schema (generally done by the `valid_elements`, `valid_children` or `extended_valid_elements` config). This will likely need mentioning in the migration guide as a minor note. I'm not sure if it should be in the release notes sorry.

File: modules/ROOT/pages/questions/question-14.adoc (hunk 15 above)
Comment (lnewson) [reply modules/ROOT/pages/questions/question-14.adoc:1]: I think Mike has answered this already. There's also some details related to this in my reply to Q13.

File: modules/ROOT/pages/questions/question-15.adoc (hunk 16 above)
Comment (lnewson) [reply modules/ROOT/pages/questions/question-15.adoc:1]: See the answer to Q13, but the gotcha is you need to make sure the schema is configured to allow the elements/attributes used in the media elements now as they can no longer be treated as "special" (where the contents is ignored) since we're using the native [DOMParser API](https://developer.mozilla.org/en-US/docs/Web/API/DOMParser) behind the scenes now.

File: modules/ROOT/pages/questions/question-13.adoc (hunk 14 above)
Comment (TheSpyder) [reply modules/ROOT/pages/questions/question-13.adoc:1]: Also worth noting the parser we switched to is not our own, under the hood our `DomParser` API uses the browser parsing API of the same name
https://developer.mozilla.org/en-US/docs/Web/API/DOMParser

Comment (lnewson) [reply modules/ROOT/pages/questions/question-13.adoc:1]: Yeah I missed that here, but had mentioned that in another answer sorry (see Q15).

File: modules/ROOT/pages/questions/question-17.adoc (hunk 18 above)
Comment (ltrouton) [reply modules/ROOT/pages/questions/question-17.adoc:1]: These option names were changed to make them more consistent with the other `link` and `autolink` options available. They are purely name changes. The functionality of these options and the possible values they can be given has not changed

File: modules/ROOT/pages/questions/question-18.adoc (hunk 19 above)
Comment (ltrouton) [reply modules/ROOT/pages/questions/question-18.adoc:1]: > Should this be documented entirely separately or is it safe to just note the re-naming also applies to `autolink`?

I am inclined to say it should be the latter. Setting the `link_default_target` option will affect both `autolink` and `link` plugins i.e. the default link target cannot be controlled separately for each of the plugins

File: modules/ROOT/pages/questions/question-21.adoc (hunk 22 above)
Comment (ltrouton) [reply modules/ROOT/pages/questions/question-21.adoc:1]: It's worth noting that the old `mceInsertTable` command still exists in v6. In v5, the command was serving two purposes: opening the insert/edit table dialog or inserting a table without using a dialog if additional arguments were provided. For example (old way):
```
tinymce.activeEditor.execCommand('mceInsertTable'); // Opens dialog
tinymce.activeEditor.execCommand('mceInsertTable', false, { rows: 2, columns: 2 }); // Inserts table into content directly without opening a dialog
```
It was not ideal that the `mceInsertTable` command was serving two purposes so the `mceInsertTableDialog` command was created and the opening dialog functionality was removed from `mceInsertTable` command. For example (new way):
```
tinymce.activeEditor.execCommand('mceInsertTableDialog'); // Opens dialog
tinymce.activeEditor.execCommand('mceInsertTable', false, { rows: 2, columns: 2 }); // Inserts table into content directly
```
So basically in the migration guide, we will need to mention that:
> The `mceInsertTable` command can no longer open the insert table dialog. Use the `mceInsertTableDialog` command instead

so integrators know to use the new command if they were previously using the `mceInsertTable` command to open the table dialog

File: modules/ROOT/pages/questions/question-19.adoc (hunk 20 above)
Comment (TheSpyder) [reply modules/ROOT/pages/questions/question-19.adoc:1]: This was discovered when we tried to support using the tabfocus plugin in RTC. The plugin had long since not done much of anything; it was designed for older browsers. The entire plugin was replaced with tabindex on the editor content in TINY-8315.

This did indeed happen completely independently of the rename and deprecation efforts, those should have been cancelled when we decided to remove it.

File: modules/ROOT/pages/questions/question-08.adoc (hunk 9 above)
Comment (EkimChau) [reply modules/ROOT/pages/questions/question-08.adoc:1]: - textpatterns is no longer a plugin and is now part of TinyMCE core
- the previous plugin configuration to enable textpatterns is no longer required { plugins: 'textpattern' } and should be removed from the plugin configuration

more info @metricjs 
- migration - steps to migrate 5x textpattern to 6x
- configuration how to enable, disable textpattern functionality
- default configuration of textpattern

Comment (TheSpyder) [reply modules/ROOT/pages/questions/question-08.adoc:1]: `textpattern_patterns` is a configuration option, and I believe `text_patterns` is equivalent. Upgraders will need to rename their configuration and (as Mike said) remove `textpattern` from their plugins list. No other changes should be required.

File: modules/ROOT/pages/questions/question-05.adoc (hunk 6 above)
Comment (metricjs) [reply modules/ROOT/pages/questions/question-05.adoc:1]: Bringing it into line with PowerPaste, and true seeming to be more useful, were the reasons I remember, yes.

File: modules/ROOT/pages/questions/question-08.adoc (hunk 9 above)
Comment (metricjs) [reply modules/ROOT/pages/questions/question-08.adoc:1]: Yes, here's the changes:
- textpattern plugin was moved to core, so it should not be specified in the `plugins` option anymore, and it is essentially "on by default"
- `textpattern_patterns` was renamed to `text_patterns`
- you can now set `text_patterns: false` or `text_patterns: []` to turn the functionality off
- the textpattern plugin's API was removed

File: modules/ROOT/pages/questions/question-08.adoc [hunk 25]
```diff
@@ -0,0 +1,12 @@
...
+from being a plugin to being part of Core:
+
+   * `textpattern_patterns` became `text_patterns`
+
+   Socratic question: is this re-named item a *Function* or a
```
Comment (metricjs) [inline modules/ROOT/pages/questions/question-08.adoc:6]: `text_patterns` is an Option

File: modules/ROOT/pages/questions/question-08.adoc [hunk 26]
```diff
@@ -0,0 +1,12 @@
...
+   *Command*?
+
+   Does this re-naming constitute something an upgrader needs to be
+   aware of? That is, does the re-naming need to be noted in the
+   _Migration Guide_? (And, if so, what does the upgrader need to do
+   to cater to the change?)
```
Comment (metricjs) [inline modules/ROOT/pages/questions/question-08.adoc:12]: Implied above, but the upgrader will need to rename the option in their config.

File: modules/ROOT/pages/questions/question-09.adoc [hunk 27]
```diff
@@ -0,0 +1,14 @@
...
+   > `text_patterns: []` will turn the functionality off
+
+   Is this a change from previously?
+
+   Or, if not, has the means of turning the functionality off (if
+   desired) changed from previously?
+
+   And, if it has changed, does this change constitute something an
+   upgrader needs to be aware of? That is, does this change need to be
+   noted in the _Migration Guide_?
```
Comment (metricjs) [inline modules/ROOT/pages/questions/question-09.adoc:14]: Apologies, this doubles up with my reply above, but:

Before, if you wanted Text Pattern functionality you had to opt in by including it as a plugin. Now, it's part of core, so the functionality is included and "on" by default. In order to opt out, you now need to use the `text_patterns` option, setting it to either `false` or `[]`.  

So yes, will need to be in the migration guide.

File: modules/ROOT/pages/questions/question-12.adoc (hunk 13 above)
Comment (metricjs) [reply modules/ROOT/pages/questions/question-12.adoc:1]: a. No

b. I think yes, as if someone does contact us about it, at least then we can know what it actually did. Better than us guessing, I think.

File: modules/ROOT/pages/questions/question-13.adoc (hunk 14 above)
Comment (metricjs) [reply modules/ROOT/pages/questions/question-13.adoc:1]: Haven't read all the questions, so unsure if this is mentioned elsewhere, but: generally, I think it's worth mentioning the parser change in the Release Notes due to the reason we did it - which is security improvements. Regarding this particular aspect - the relationship between the parser and the media plugin - that I don't think we need to make particular note of.

File: modules/ROOT/pages/questions/question-16.adoc (hunk 17 above)
Comment (metricjs) [reply modules/ROOT/pages/questions/question-16.adoc:1]: I can answer this one for Hannes - yes, it was done for clarity and consistency. Options generally follow a naming convention that means they are prefixed with the name of the plugin they belong to. These options did not follow that convention, though it is an odd situation due to them belonging to two plugins. But we wanted to rename them to make it clearer what they were for, anyway.

File: modules/ROOT/pages/questions/question-19.adoc (hunk 20 above)
Comment (metricjs) [reply modules/ROOT/pages/questions/question-19.adoc:1]: I already wrote Release Notes and Migration Guide copy for this. I'll try to track it down...

Comment (metricjs) [reply modules/ROOT/pages/questions/question-19.adoc:1]: This was for the old markdown-based docs but the PR is here: https://github.com/tinymce/tinymce-docs/pull/2183

File: modules/ROOT/pages/questions/question-22.adoc (hunk 23 above)
Comment (metricjs) [reply modules/ROOT/pages/questions/question-22.adoc:1]: Let's not mention anything about future plans. For now, this is purely part of modernising TinyMCE's default behaviour.
